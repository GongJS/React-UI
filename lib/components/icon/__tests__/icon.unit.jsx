import * as renderer from 'react-test-renderer';
import React from 'react';
import Icon from '../Icon';
import { mount } from 'enzyme';

// 注意: 运行icon测试用例的时候,先别导入icnfont.js !!!
describe('<Icon />', () => {
  it('renders icon component', () => {
    const json = renderer.create(<Icon name={'alipay'} height={'20px'}/>).toJSON();
    expect(json).toMatchSnapshot();
  });
  it('icon component click events', () => {
    const fn = jest.fn();
    const component = mount(<Icon name="qq" onClick={fn}/>);
    component.find('svg').simulate('click');
  })
});
