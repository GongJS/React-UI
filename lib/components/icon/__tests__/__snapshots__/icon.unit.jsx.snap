// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Icon /> renders icon component 1`] = `
<svg
  className="r-icon"
  style={
    Object {
      "height": "20px",
      "width": "1.2em",
    }
  }
>
  <use
    fill="#3963bc"
    xlinkHref="#icon-alipay"
  />
</svg>
`;
